### PR TITLE
kfence: Address Jann's v6 comments

### PR DESCRIPTION
**Pull request by melver**

### CODE REVIEW

File: arch/x86/mm/fault.c [hunk 1]
```diff
@@ -726,7 +726,8 @@ no_context(struct pt_regs *regs, unsigned long error_code,
 	if (IS_ENABLED(CONFIG_EFI))
 		efi_recover_from_page_fault(address);
 
-	if (kfence_handle_page_fault(address))
+	/* Only not-present faults should be handled by kfence. */
```
Comment (ramosian-glider) [inline arch/x86/mm/fault.c:729]: Nit: KFENCE?

Comment (melver) [reply arch/x86/mm/fault.c]: Done.

File: mm/kfence/core.c [hunk 2]
```diff
@@ -225,13 +225,22 @@ static __always_inline void for_each_canary(const struct kfence_metadata *meta,
...
+	/*
+	 * We'll iterate over each canary byte per-side until fn() returns
+	 * false. However, we'll still iterate over the canary bytes on the
+	 * right of the object even if there was an error in the canary bytes on
+	 * the left of the object. Specifically, if check_canary_byte()
+	 * generates an error, showing both sides might give more clues as to
+	 * what the error is about when displaying which bytes were corrupted.
+	 */
+
+	/* Apply to left of object. */
```
Comment (ramosian-glider) [inline mm/kfence/core.c:237]: Nit: "to the left"/"to the right"?

**Review by ramosian-glider — APPROVED** [review]
LGTM, sorry for the delay!

File: mm/kfence/core.c (hunk 2 above)
Comment (melver) [reply mm/kfence/core.c:237]: Done

"on" to "to" both is fine here, but I think for consistency I'll switch back to "to".

**Pull request merged**